### ISSUE TITLE
Namespace for jingle transport candidates

### ISSUE THREAD

**Issue by arudnev** [legastero/stanza#411]
While troubleshooting issue with jingle sessions between 12.x and 11.x based clients when TURN is involved I've noticed mismatch of xml namespaces for transport and candidate elements coming from  newer client. Not sure yet if this has anything to do with broken connectivity between older 11.x and newer 12.x based clients when relayed connection is required, but going to mention it just in case it could be reason for compatibility issues with other implementations.

It seems that according to [XEP-0176](https://xmpp.org/extensions/xep-0176.html) jingle/content/transport that has xmlns='urn:xmpp:jingle:transports:ice-udp:1' should have nested candidate in the same namespace

```xml
<transport xmlns='urn:xmpp:jingle:transports:ice-udp:1' ...>
   <candidate .../>
</transport>
```

and in fact in candidates coming from clients that use older 11.x version of stanza library that's how it looks.

When using 12.4.3 version I see the following in outgoing transport-info actions:

```xml
<transport xmlns="urn:xmpp:jingle:transports:ice-udp:1" ...>
  <candidate xmlns="urn:xmpp:jingle:transports:ice:0" .../>
</transport>
```

It seems that new xml namespace used in the candidates comes from currently deferred [XEP-0371](https://xmpp.org/extensions/xep-0371.html) that is intended to supersede [XEP-0176](https://xmpp.org/extensions/xep-0176.html), but transport structure in [XEP-0371](https://xmpp.org/extensions/xep-0371.html) is defined as following:

```xml
<transport xmlns='urn:xmpp:jingle:transports:ice:0' ...>
  <candidate .../>
</transport>
```

I have not looked into how JXT (especially implementation used with older 11.x library) treats this mismatch of xml namespace between transport and candidate elements, but it seems that due to this mismatch resulting xml can cause various issues with mapping from xml to json (or other native data representations) depending on how mapping framework uses xml schemas / namespaces defined in those XEPs.

**Comment by arudnev** [comment]
It seems that the issue is not related to use of relayed connection and most likely can be reproduced by establishing p2p session between older 11.x and newer 12.x based clients.

It comes down to the fact that older 11.x clients only had JXT definitions for `xmlns="urn:xmpp:jingle:transports:ice-udp:1"`, so when such client gets transport-info with candidates in `xmlns="urn:xmpp:jingle:transports:ice:0"` they are not mapped to json.

Reversing order of ice JXT definitions (or monkey-patching `JXT.Registry.define` to do reordering of defs with those two namespaces) makes 12.x clients compatible with older 11.x clients (as long as default value of `NS_JINGLE_ICE_UDP_1` for `ICESession.transportType` is being used in 12.x clients)

https://github.com/legastero/stanza/blob/59fd14a7a45d678920d585b64c96dced45e5943b/src/protocol/xep0176.ts#L110

I presume the issue is that the first translator for given path is being used when candidate is converted to xml  instead of taking namespace of parent transport element into account, but unfortunately I can not quickly figure out how to refine definitions or some conversion logic to address that.

**Comment by legastero** [comment]
This should be fixed now in 12.4.4.

Thanks!
